testplan/testing/multitest/base.py

```diff
@@ -804,7 +804,7 @@ def _run_suite_related(self, testsuite, method):
         return method_report
 
     def _run_case_related(self, method, testcase, case_result):
-        interface.check_signature(method, ["name", "self", "env", "result"])
+        interface.check_signature(method, ["self", "name", "env", "result"])
         method(testcase.__name__, self.resources, case_result)
```

Comment:
but we didn't pass self here, should be a problem.

Reply:
_run_case_related is always used with the per_testcase and post_testcase methods, they are always looked up from the testsuite instence
```python
        pre_testcase = getattr(testsuite, "pre_testcase", None)
        post_testcase = getattr(testsuite, "post_testcase", None)
```
which is a bound method, so running it will implicitly get the instance (aka: self) as the first argument. so no need to pass.

Please note, that this is when the testsuite class has a `pre_testcase` or `post_testcase` method, which is not much documented and is different from the `@pre_tescase` `@post_testcase` decorator for a testsuite, which expect not bound functions, that taking `"name", "self", "env", "result"` 

```python
   @wraps(testcase_method)
    def testcase_with_pre(*args, **kwargs):
        """
        Testcase with prelude
        """
        for prelude in preludes:
            prelude(
                testcase_method.__name__,
                *args,
                **_get_kwargs(kwargs, testcase_method)
            )
        return testcase_method(*args, **kwargs)
```

None of the unittests testing the member function way of doing this, that is why the above signature check never found.